Purchases/Networking/Caching/CallbackCache.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  CallbackCache.swift
+//
+//  Created by Joshua Liebowitz on 11/18/21.
+
+import Foundation
+
+/**
+ Generic callback cache whose primary usage is to help ensure API calls in flight are not duplicated.
+ Users of this class will have determined that an API call is running and they will store a completion block from
```

Comment:
"Users of this class will have determined that an API call is running" - do we not use this class for the very first call?

Reply:
Good point, we do use this first, too.

Reply:
Updated comment.